app/common/UserAPI.ts

```diff
@@ -1112,6 +1112,22 @@ export class DocAPIImpl extends BaseAPI implements DocAPI {
   }
 }
 
+interface DocWorkerInfo {
+  docWorkerUrl: string|null;
+  internalDocWorkerUrl: string|null;
+  selfPrefix?: string;
+}
+
+function getUrlFromPrefix(homeUrl: string, prefix?: string) {
+  if (!prefix) {
+    // This should never happen.
+    throw new Error('missing selfPrefix for docWorkerUrl');
+  }
```

Comment:
So why it's an option in the params of the function ?

Reply:
Good question: it's because it's an internal function used to factorize code between `getPublicDocWorkerUrl` and `getInternalDocWorkerUrl`.

This function is called when fallbacking, and in that case the prefix should be provided.

So the parameter is only optional to fit with the signature of the two public functions.

Reply:
I made something that should answer to your remark